x-pack/plugins/observability_solution/infra/server/routes/entities/get_latest_entity.ts

```diff
@@ -5,64 +5,55 @@
  * 2.0.
  */
 
-import { ENTITY_LATEST, entitiesAliasPattern } from '@kbn/entities-schema';
-import { type EntityClient } from '@kbn/entityManager-plugin/server/lib/entity_client';
-import { ENTITY_TYPE, SOURCE_DATA_STREAM_TYPE } from '@kbn/observability-shared-plugin/common';
-import type { ObservabilityElasticsearchClient } from '@kbn/observability-utils-server/es/client/create_observability_es_client';
+import type { EntityClient } from '@kbn/entityManager-plugin/server/lib/entity_client';
 import type { Logger } from '@kbn/logging';
-
-const ENTITIES_LATEST_ALIAS = entitiesAliasPattern({
-  type: '*',
-  dataset: ENTITY_LATEST,
-});
+import { isArray } from 'lodash';
 
 interface EntitySourceResponse {
   sourceDataStreamType?: string | string[];
 }
 
 export async function getLatestEntity({
-  inventoryEsClient,
   entityId,
   entityType,
   entityManagerClient,
+  from,
+  to,
   logger,
 }: {
-  inventoryEsClient: ObservabilityElasticsearchClient;
   entityType: 'host' | 'container';
   entityId: string;
   entityManagerClient: EntityClient;
+  from: string;
+  to: string;
   logger: Logger;
 }): Promise<EntitySourceResponse | undefined> {
   try {
-    const { definitions } = await entityManagerClient.getEntityDefinitions({
-      builtIn: true,
+    const entityDefinitionsSource = await entityManagerClient.v2.readSourceDefinitions({
       type: entityType,
     });
 
-    const hostOrContainerIdentityField = definitions[0]?.identityFields?.[0]?.field;
+    const hostOrContainerIdentityField = entityDefinitionsSource[0]?.identity_fields?.[0];
     if (hostOrContainerIdentityField === undefined) {
       return undefined;
     }
 
-    const response = await inventoryEsClient.esql<
-      {
-        'source_data_stream.type'?: string | string;
-      },
-      { transform: 'plain' }
-    >(
-      'get_latest_entities',
-      {
-        query: `FROM ${ENTITIES_LATEST_ALIAS}
-        | WHERE ${ENTITY_TYPE} == ?
-        | WHERE ${hostOrContainerIdentityField} == ?
-        | KEEP ${SOURCE_DATA_STREAM_TYPE}
-      `,
-        params: [entityType, entityId],
-      },
-      { transform: 'plain' }
-    );
+    const entities = await entityManagerClient.v2.searchEntities({
```

Comment:
Hi @klacabane, I wanted to check with you about our usage of start/end params in `searchEntities` here: I saw they are required and type as `string` so I guess in our case here we want them to be the same as the date picker on the page, right? So for example we will pass them in that format: data picker with 1 hour selected as time range: `start: '2024-12-11T10:49:15Z', end: '2024-12-11T11:49:15Z',`